lldb/bindings/interface/SBThreadExtensions.i

```diff
@@ -51,6 +51,12 @@ STRING_EXTENSION_OUTSIDE(SBThread)
                 for idx in range(self.GetStopReasonDataCount())
             ]
 
+        def set_selected_frame(self, frame):
+            if isinstance(frame, SBFrame):
+                self.SetSelectedFrame(frame.idx)
```

Comment:
Maybe it should check that the frame really belongs to this thread before selecting a completely unrelated frame?

Reply:
Callers shouldn't have to do that; StackFrames know their Threads, so SetSelectedFrame should really reject this (though in that case it ought to also return an error.)

Reply:
SetSelectedFrame returns the index of the selected frame.  It should check the thread and if it doesn't match, return LLDB_INVALID_FRAME_ID.

Reply:
good call, done.

Reply:
oh my reply was to @labath, before I saw @jimingham's comments.

Reply:
I've updated this helper method to check the `SBFrame` argument value belongs to the target instance of `SBThread`. If it doesn't, a `ValueError` is raised. Does that work for you @jimingham?